### PR TITLE
[AKS] Add support of creating private cluster

### PR DESCRIPTION
**Pull request by feiskyer**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[AKS] aks create: add `--enable-private-cluster`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by feiskyer** [comment]
@yonzhan could you help adding this to S166?

**Comment by yungezz** [comment]
HI @fengzhou-msft @qwordy  could you pls look at the PR? thanks.

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -18,6 +18,7 @@ az webapp create : Fixing issue when running the command with --runtime
 **ACS**
 
 * aks create/update: add `--vnet-subnet-id` validation
+* aks create: add `--enable-private-cluster`
```
Comment (zqingqing1) [inline src/azure-cli/HISTORY.rst]: I think we should have separate section for aks, like:
**AKS**
* aks create: add `--enable-private-clister`

Comment (feiskyer) [reply src/azure-cli/HISTORY.rst]: tests are added in `TestPopulateApiServerAccessProfile`

Comment (feiskyer) [reply src/azure-cli/HISTORY.rst]: **ACS** is actually "AKS", our module name is "acs" in Azure CLI

Comment (feiskyer) [reply src/azure-cli/HISTORY.rst]: let me rename it to AKS anyway.

Comment (fengzhou-msft) [reply src/azure-cli/HISTORY.rst]: The history note is added in a released version 2.1.0. Please delete it and use PR title or PR description, they will be included in history.rst before release. You can check the guide [here](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

Comment (fengzhou-msft) [reply src/azure-cli/HISTORY.rst]: The history note is added in a released version 2.1.0. You don't need to manually add history now, please delete it and use PR title or PR description, they will be included in history.rst before release. You can check the guide [here](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

Comment (feiskyer) [reply src/azure-cli/HISTORY.rst]: OK, let me delete it

Comment (feiskyer) [reply src/azure-cli/HISTORY.rst]: @fengzhou-msft deleted the changes in history. PTAL.

Comment (fengzhou-msft) [reply src/azure-cli/HISTORY.rst]: Ok. I added the history note in description since current PR title has different wordings.

**Comment by feiskyer** [comment]
@fengzhou-msft Thanks

**Pull request merged**